warehouse/query-core/src/main/java/datawave/query/planner/scanhints/IvaratorScanHint.java

```diff
@@ -0,0 +1,38 @@
+package datawave.query.planner.scanhints;
+
+import org.apache.commons.jexl3.parser.JexlNode;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.NodeTypeCount;
+
+/**
+ * Detect ivarators and set the executor scan hint to ivarator
+ */
+public class IvaratorScanHint extends ExecutorScanHintRule {
+    @Override
+    public boolean isChainable() {
+        return true;
+    }
+
+    @Override
+    public String getTable() {
+        return "shard";
```

Comment:
This table name should be configurable. While "shard" is the default shard table name (see the TableName constants), there's nothing stopping a user from creating a shard table with the name like "datawave.shard" or "test.shard". In those cases scan hints wouldn't be applied.

I don't consider this a blocking issue, feel free to do this later.